src/LCM2ROSControl.cpp

```diff
@@ -221,12 +221,64 @@ namespace valkyrie_translator
     << imu_hw_claims.size() << " IMUs" << std::endl
     << effort_hw_claims.size() << " effort-controlled joints" << std::endl
     << position_hw_claims.size() << " position-controlled joints" << std::endl);
+
+  if (!loadBehaviorGainOverrides(effortNames, controller_nh)) {
+    ROS_ERROR("Failed to load behavior gain overrides");
+    return false;
+  }
+
+  return true;
+}
+
+bool LCM2ROSControl::loadBehaviorGainOverrides(const std::vector<std::string>& effort_names, const ros::NodeHandle& controller_nh) {
+  XmlRpc::XmlRpcValue gain_overrides;
+
+  if (!controller_nh.getParam("gain_overrides", gain_overrides)) {
+    ROS_ERROR("Could not get gain_overrides, exiting");
+    return false;
+  }
+
+  std::vector<std::pair<Behavior, std::string> > behaviors = {{FREEZE, "freeze"}, {POSITION_CONTROL, "position_control"}};
+  for (auto iter = behaviors.begin(); iter != behaviors.end(); ++iter) {
```

Comment:
Another minor thing, but I would give preference to range based for loops for readability these days (see e.g. http://stackoverflow.com/a/6963910)